src/js/components/tasks/steps.tsx

```diff
@@ -0,0 +1,77 @@
+import classNames from 'classnames';
+import i18n from 'i18next';
+import _ from 'lodash';
+import React, { useEffect, useState } from 'react';
+
+import { Task } from '@/store/tasks/reducer';
+import { TASK_STATUSES } from '@/utils/constants';
+
+interface TaskStatusPathProps {
+  task: Task;
+}
+
+const TaskStatusSteps = ({ task }: TaskStatusPathProps) => {
+  const [completedSteps, setcompletedSteps] = useState<number[]>([]);
+  const [nextSteps, setNextSteps] = useState<number[]>([]);
+
+  useEffect(() => {
+    const inProgressDone = task.has_unmerged_commits ? [0, 1] : [0, 1, 2, 3];
+    const inProgressNext = task.has_unmerged_commits ? [3] : [4];
+    let done: number[], next: number[];
+    switch (task.status) {
+      case TASK_STATUSES.PLANNED:
+        done = [];
+        next = [0, 1];
+        break;
+      case TASK_STATUSES.IN_PROGRESS:
+        done = inProgressDone;
+        next = inProgressNext;
+        break;
+      case TASK_STATUSES.COMPLETED:
+        done = [0, 1, 2, 3, 4, 5]; // or maybe all the indexes of steps ?
+        next = [];
+        break;
+    }
+    setcompletedSteps(done);
+    setNextSteps(next);
```

Comment:
@jgerigmeyer I think there is something missing here, that I'm sure  we talked about before. Right now it's just checking for any uncaptured changes, but I think there should also be different steps showing as queued or done while a PR is in progress and under review? (like if there is a reviewer assigned to the QA org).